cmd/readPipelineEnv.go

```diff
@@ -45,3 +83,56 @@ func runReadPipelineEnv() error {
 
 	return nil
 }
+
+func encrypt(secret, inBytes []byte) ([]byte, error) {
+	// use SHA256 as key
+	key := sha256.Sum256(secret)
+	block, err := aes.NewCipher(key[:])
+	if err != nil {
+		return nil, fmt.Errorf("failed to create new cipher: %v", err)
+	}
+
+	// Make the cipher text a byte array of size BlockSize + the length of the message
+	cipherText := make([]byte, aes.BlockSize+len(inBytes))
+
+	// iv is the ciphertext up to the blocksize (16)
+	iv := cipherText[:aes.BlockSize]
+	if _, err = io.ReadFull(rand.Reader, iv); err != nil {
+		return nil, fmt.Errorf("failed to init iv: %v", err)
+	}
+
+	// Encrypt the data:
+	stream := cipher.NewCFBEncrypter(block, iv)
+	stream.XORKeyStream(cipherText[aes.BlockSize:], inBytes)
+
+	// Return string encoded in base64
+	return []byte(base64.StdEncoding.EncodeToString(cipherText)), err
+}
+
+// retrieve step metadata
+func readPipelineEnvMetadata() config.StepData {
+	var theMetaData = config.StepData{
+		Metadata: config.StepMetadata{
+			Name: "readPipelineEnvMetadata",
+		},
+		Spec: config.StepSpec{
+			Inputs: config.StepInputs{
+				Parameters: []config.StepParameters{
+					{
+						Name: "secret",
+						ResourceRef: []config.ResourceReference{
+							{
+								Name: "cpeSecret",
+								Type: "vaultSecret",
+							},
+						},
+						Type:      "string",
+						Mandatory: false,
+						Default:   os.Getenv("PIPER_pipelineEnv_SECRET"),
```

Comment:
we want to **reuse** token/secret. So, this part needs to be adjusted accordingly.

The[ artifactPrepareVersion uses token](https://github.com/SAP/jenkins-library/blob/97edad076ad022a6495bf5f9208573da0ef92c6d/resources/metadata/artifactPrepareVersion.yaml#L248), we could reuse it to avoid creating another token for read(write)PipelineEnv.

but artifactPrepareVersion by default uses the githubHttpsCredential as the path to the password (token) in Vault, and the path can be **changed** by the user. If it is changed, how the readPipelineEnv will know about it? Moreover, Hyperspace Onboarding generates config.yaml with changed gitHttpCredential path (for artifactPrepareVersion).
Can we implement read(write)Pipeline Env reads artifactPrepareVersion config to know githubHttpsCredential value?

Do not hesitate to include @anilkeshav27 to review process and ask questions to the team.